tests/unit/hook_packages/terraform/test_layer_linking.py

```diff
@@ -0,0 +1,43 @@
+from unittest import TestCase
+
+from parameterized import parameterized
+
+from samcli.hook_packages.terraform.hooks.prepare.layer_linking import _clean_references_list
+
+
+class TestLayerLinking(TestCase):
+    @parameterized.expand(
```

Comment:
can you add a case that contain an instance for a block defined with for_each. https://www.terraform.io/language/meta-arguments/for_each#referring-to-instances
something like `([aws_lambda_layer_version.layer1["key1"].arn, aws_lambda_layer_version.layer1["key1"], aws_lambda_layer_version.layer1)`